test/dynamo/test_export.py

```diff
@@ -3009,12 +2999,11 @@ def false_fn(x):
             return (x, x)
 
         def f_mismatch_return_length(x):
-            return cond(torch.tensor(100), true_fn, false_fn, [x])
+            return cond(torch.tensor(True), true_fn, false_fn, [x])
```

Comment:
We should... probably just allow values that are coercible to bool. Otherwise it sounds like we might break some things (do we end up breaking anything internal?)

Reply:
I don't really know if there's a clean way to do this

Reply:
easiest way is probably just to validate that `pred` is a python number type or a scalar Tensor (we don't need to check the dtype)

Reply:
em... I don't know how large chance we'll break internal tests. It also sound good to me to not check the dtype. Maybe let's just change it back to original behavior that we allow boolean casting in predicate